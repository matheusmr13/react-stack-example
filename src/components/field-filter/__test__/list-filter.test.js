import React from 'react';
import ListFilter from 'components/field-filter/list-filter';
import { shallow } from 'enzyme';

describe('ListFilter component', () => {
	it('should render properly', () => {
		const spec = {
			id: 'myId',
			name: 'List field',
			values: []
		};
		const listFilter = shallow(<ListFilter spec={spec} />);
		expect(listFilter).toMatchSnapshot();
	});
});
