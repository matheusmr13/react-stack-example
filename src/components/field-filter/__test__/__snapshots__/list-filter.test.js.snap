// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListFilter component should render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListFilter
    spec={
        Object {
            "name": "List field",
            "values": Array [],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "anchorOrigin": Object {
        "horizontal": "left",
        "vertical": "bottom",
      },
      "animated": true,
      "dataSource": Array [],
      "dataSourceConfig": Object {
        "text": "name",
      },
      "disableFocusRipple": true,
      "filter": [Function],
      "floatingLabelText": "List field",
      "fullWidth": true,
      "menuCloseDelay": 300,
      "onNewRequest": [Function],
      "onUpdateInput": [Function],
      "open": false,
      "openOnFocus": true,
      "targetOrigin": Object {
        "horizontal": "left",
        "vertical": "top",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "anchorOrigin": Object {
          "horizontal": "left",
          "vertical": "bottom",
        },
        "animated": true,
        "dataSource": Array [],
        "dataSourceConfig": Object {
          "text": "name",
        },
        "disableFocusRipple": true,
        "filter": [Function],
        "floatingLabelText": "List field",
        "fullWidth": true,
        "menuCloseDelay": 300,
        "onNewRequest": [Function],
        "onUpdateInput": [Function],
        "open": false,
        "openOnFocus": true,
        "targetOrigin": Object {
          "horizontal": "left",
          "vertical": "top",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
